service/history/shard/context_impl.go

```diff
@@ -367,7 +367,20 @@ func (s *ContextImpl) UpdateQueueAckLevel(
 			ClusterAckLevel: make(map[string]int64),
 		}
 	}
-	s.shardInfo.QueueAckLevels[category.ID()].AckLevel = convertTaskKeyToAckLevel(category.Type(), ackLevel)
+	persistenceAckLevel := convertTaskKeyToPersistenceAckLevel(category.Type(), ackLevel)
+	s.shardInfo.QueueAckLevels[category.ID()].AckLevel = persistenceAckLevel
+
+	// if cluster ack level is less than the overall ack level, update cluster ack level
+	// as well to prevent loading too many tombstones if the cluster ack level is used later
+	// this may happen when adding back a removed cluster or rolling back the change for using
+	// single queue in timer/transfer queue processor
+	if queueAckLevel, ok := s.shardInfo.QueueAckLevels[category.ID()]; ok {
```

Comment:
this ok check is redundant because line above already access that key in the map. 
could assign queueAckLevel := ... directly.